File: csharp/O/After/Strategy/FuseDCStrategy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace After.Strategy;

internal class FuseDCStrategy : IProtectionStrategy
{
    public double CalcBreakingCapacity()
    {
        return 4;
    }

    public double CalcIn()
    {
        return 4;
    }

    public bool CheckProtection()
    {
        return true;
    }
}

File: js-ts/O/after/electrical-protection/Strategy/FuseDCStrategy.ts
// A concrete strategy for a DC Fuse.
import { IProtectionStrategy } from "./IProtectionStrategy.ts";

export class FuseDCStrategy implements IProtectionStrategy {
    public calcIn(In: number): number {
        return In + 4;
    }

    public calcBreakingCapacity(): number {
        return 4;
    }

    public checkProtection(): boolean {
        return true;
    }
}
